### ISSUE TITLE
New config file, training scripts for complex networks.

### ISSUE THREAD

**Issue by VenuGopalVasarla** [VenuGopalVasarla/DCGAN-pytorch-venu#14]
This needs to be closed only after complete testing and review.
Mainly aimed at stable output for custom dataset.

- [x] Add new training  scripts.
- [ ] New loss function.
- [ ] Construct different mini-batches for real and fake, i.e. each mini-batch needs to contain only all real images or all generated images

- [ ] Avoid Sparse Gradients: ReLU, MaxPool. Use: Leaky ReLU

- [ ] For Downsampling, use: Average Pooling, Conv2d + stride

- [ ] For Upsampling, use: PixelShuffle, ConvTranspose2d + stride

- [ ] Label Smoothing, i.e. if you have two target labels: Real=1 and Fake=0, then for each incoming sample, if it is real, then replace the label with a random number between 0.7 and 1.2, and if it is a fake sample, replace it with 0.0 and 0.3 (for example).

- [ ] Use SGD for discriminator and ADAM for generator.

- [ ] Train discriminator more times.

- [ ] use dropouts in the generator.

- [ ] Write new config file for the specifications of networks, trainer... etc
